src/redisraft.c

```diff
@@ -757,9 +782,28 @@ static void handleRedisCommandAppend(RedisRaftCtx *rr,
         return;
     }
 
-    RaftReq *req = RaftReqInit(ctx, RR_REDISCOMMAND);
+    long long timeout = 0;
+    bool blocking = false;
+    if (cmd_flags & CMD_SPEC_BLOCKING) { /* protect against blocking commands in a MULTI above */
+        blocking = true;
+        long double tmp;
+        if (RaftRedisExtractBlockingTimeout(cmds, &tmp) != REDISMODULE_OK) {
```

Comment:
Suggest combining `RaftRedisExtractBlockingTimeout` with `validTimeout`, same as redis does at `getTimeoutFromObjectOrReply`. Anyway, if you still want to keep them both,`validTimeout` should be renamed, because it also replies to the client.